benchmarking_hallucination_metrics/benchmark_hallucination_metrics.ipynb

```diff
@@ -0,0 +1,4732 @@
+        "outputId": "e4ea11e1-387e-4d57-ed02-238d48250a26"
+      },
+      "outputs": [],
+      "source": [
+        "SHOULD_RUN = False\n",
+        "\n",
+        "if SHOULD_RUN:\n",
+        "  res_dict = {}\n",
+        "  for dataset_name in unique_datasets:\n",
+        "    print(dataset_name)\n",
+        "    try:\n",
+        "      res_dict[dataset_name] = score_dataset_with_metrics(data[data.source_ds == dataset_name])\n",
+        "      res_dict[dataset_name].to_csv(f\"results_{dataset_name}.csv\")\n",
+        "    except:\n",
+        "      print(f\"Error with dataset {dataset_name}\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "The above code may take a long time to run. Once you do have results, you can plot them with the following code.\n",
+        "\n",
+        "We'll use our small example to illustrate:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "\n",
+        "res_dict[dataset_name]"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 33,
+      "metadata": {
+        "id": "JSaO0CyV_Unz"
+      },
+      "outputs": [],
+      "source": [
+        "from sklearn.metrics import roc_curve, roc_auc_score, auc\n",
+        "import matplotlib.pyplot as plt\n",
+        "\n",
+        "IMPUTATION_VALUE = .5\n",
+        "\n",
+        "def calculate_and_plot_auroc(result_df, dataset_name):\n",
+        "    result = result_df\n",
+        "    score_list = [\n",
+        "        (\"TLM\", result[\"tlm_score\"]),\n",
+        "        (\"RAGAS Faithfulness\", result[\"ragas_faithfulness\"]),\n",
+        "        (\"RAGAS Answer Relevancy\", result[\"ragas_answer_relevancy\"]),\n",
+        "        # Uncomment this if you computed Faithfulness with HHEM\n",
+        "        # (\"RAGAS Faithfulness HHEM\", result[\"ragas_faithfulness_hhem\"]),\n",
+        "        (\"RAGAS++ Faithfulness\", result[\"ragas++_faithfulness_improved\"]),\n",
+        "        (\"RAGAS++ Answer Relevancy\", result[\"ragas++_answer_relevancy_improved\"]),\n",
+        "        (\"G-eval Correctness\", result[\"geval_correctness\"]),\n",
+        "        (\"Self-Evaluation\", result[\"self_evaluation_score\"])\n",
+        "    ]\n",
+        "    result[\"is_correct\"] = result.label == \"PASS\"\n",
+        "    for name, score in score_list:\n",
+        "      fpr, tpr, _ = roc_curve(result.is_correct, score.fillna(IMPUTATION_VALUE))\n",
+        "      auroc = auc(fpr, tpr)\n",
+        "      if any(score.isnull()):\n",
+        "        plt.plot(fpr, tpr, label=f'{name} (auroc = {auroc:.3f})', linestyle='dotted')\n",
+        "      else:\n",
+        "        plt.plot(fpr, tpr, label=f'{name} (auroc = {auroc:.3f})')\n",
+        "\n",
+        "\n",
+        "\n",
+        "    plt.legend(loc='lower right', bbox_to_anchor=(1, 0))\n",
```

Comment:
Is there a reason you are keeping the legend in the plot itself? Imo it'll be better to move the legend to the side (it's taking up/covering a huge chunk of the actual plot). 

This should work:
```plt.legend(bbox_to_anchor=(1, 1))```

Reply:
Yeah, there's two ways to do this. 
In one way, the legend goes to the side. This makes it all very clear but I find it visually distracting to have the legend way over to the right - it makes the image very wide

<img width="721" alt="Screenshot 2024-09-06 at 3 02 14 PM" src="https://github.com/user-attachments/assets/e3476bc8-64ed-4135-98f9-2aef53aa650f">

The other way is to have the legend within the plot. The lower right is a natural place 
<img width="571" alt="Screenshot 2024-09-06 at 3 02 53 PM" src="https://github.com/user-attachments/assets/543a8c12-196c-4f82-ba8b-a49ed6a8ec6f">

I find that mainly the lines that are not predictive (e.g. run diagonal with auroc of .5)are covered up, which I think is fine. I agree that this sample graph is more covered than the actual metric graphs, so I'll change it!